### PR TITLE
allows user to override ServerCapabilities

### PR DESCRIPTION
**Pull request by randy3k**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,20 @@ These editors are supported by installing the corresponding package.
 
 With [lintr](https://github.com/jimhester/lintr) v2.0.0, the linters can be specified by creating the `.lintr` file at the project or home directory. Details can be found at lintr [documentation](https://github.com/jimhester/lintr#project-configuration). The option `languageserver.default_linters` is now deprecated in favor of the `.lintr` approach.
 
+### Custimizing server capbabilities
```
Comment (renkun-ken) [inline README.md]: Typo

Comment (randy3k) [reply README.md]: nice catch

**Comment by renkun-ken** [comment]
It seems that user could change those lsp options such as `TextDocumentSyncOptions` through this and languageserver will immediately have strange behavior. Should we do anything about it, e.g. add a statement or something?

**Comment by renkun-ken** [comment]
LGTM

**Pull request merged**